cms/core/tests/test_video_embed_block.py

```diff
@@ -0,0 +1,254 @@
+                {
+                    "link_url": "https://player.vimeo.com/video/test/908205163",
+                    "image": image.id,
+                    "title": "The video",
+                    "link_text": "Watch the video",
+                }
+            )
+            block.clean(value)
+
+        self.assertEqual(
+            info.exception.block_errors["link_url"].message,
+            "The link URL must use a valid vimeo or youtube video URL",
+        )
+
+    def test_videoembedblock_clean__invalid_vimeo_link_3(self):
+        """Check the VideoEmbedBlock validates the supplied URL."""
+        block = VideoEmbedBlock()
+        image = ImageFactory.create()
+
+        with self.assertRaises(StructBlockValidationError) as info:
+            value = block.to_python(
+                {
+                    "link_url": "https://vimeo.com/showcase/7934865/foo/493407585",
+                    "image": image.id,
+                    "title": "The video",
+                    "link_text": "Watch the video",
+                }
+            )
+            block.clean(value)
+
+        self.assertEqual(
+            info.exception.block_errors["link_url"].message,
+            "The link URL must use a valid vimeo or youtube video URL",
+        )
+
+    def test_videoembedblock_clean__valid_vimeo_link_1(self):
+        """Check the VideoEmbedBlock validates the supplied URL."""
+        block = VideoEmbedBlock()
+        image = ImageFactory.create()
+        value = block.to_python(
+            {
+                "link_url": "https://vimeo.com/908205163?share=copy",
+                "image": image.id,
+                "title": "The video",
+                "link_text": "Watch the video",
+            }
+        )
+        block.clean(value)
+
+        self.assertEqual(block.clean(value), value)
+
+    def test_videoembedblock_clean__valid_vimeo_link_2(self):
+        """Check the VideoEmbedBlock validates the supplied URL."""
+        block = VideoEmbedBlock()
+        image = ImageFactory.create()
+        value = block.to_python(
+            {
+                "link_url": "https://player.vimeo.com/video/908205163",
+                "image": image.id,
+                "title": "The video",
+                "link_text": "Watch the video",
+            }
+        )
+        block.clean(value)
+
+        self.assertEqual(block.clean(value), value)
+
+    def test_videoembedblock_clean__valid_vimeo_link_3(self):
+        """Check the VideoEmbedBlock validates the supplied URL."""
+        block = VideoEmbedBlock()
+        image = ImageFactory.create()
+        value = block.to_python(
+            {
+                "link_url": "https://vimeo.com/showcase/7934865/video/493407585",
+                "image": image.id,
+                "title": "The video",
+                "link_text": "Watch the video",
+            }
+        )
+        block.clean(value)
+
+        self.assertEqual(block.clean(value), value)
```

Comment:
Can you please add a few tests for the embed_url in context?

Reply:
Done.